csharp/Schleifen/Readme.md

```diff
@@ -0,0 +1,120 @@
+
+Sehen wir uns wie so eine Schleife aussieht an :
+
+```csharp
+for (int i = 0; i < 4; i = i + 1)
+{
+    Console.WriteLine(i);
+}
+```
+
+Der erste Befehl weißt einer Variable ```i``` von Typ ```int``` ein Anfangswert von 0 zu. Die Abbruchbestimmung sagt das die Schleife so lange bis das ```i``` kleiner vier ist, ausgeführt werden soll. Der letzte Parameter ist typischerweiße dazu benötigt das Wert in ```i``` zu erhöhen wie auch in diesen Beispiel. Nach jedem Durchgang wird das Wert um eins erhöht. 
+
+Eine Frage: Welche Werte werden nach Ausführung in _Console_ angezeigt?
+
+### Aufgabe
+ 1) Schreibe in einer Variable ```string[] freunde``` die Namen fünf Freunde oder Freundinnen. Gib nacheinander alle Namen in _Console_ aus, indem du die ```for``` Schleife verwendest.
+
+ 2) Gib die alle Namen in umgekehrte Reihenfolge aus. 
+## While
+
+Eine ```while``` Schleife ist eine sogenannte Kopfgesteuerte Schleife. Kopf gesteuert deswegen, weil sich die Steuerung der Schleife oben befindet.
+
+```csharp
+int anzahlVonZeichen = 0;
+while (anzahlVonZeichen < 10)
+{
+    Console.Write("X");
+    anzahlVonZeichen++;
+}
+```
+
+Im unterschied zu ```for```, hat die ```while``` Schleife nur ein Befehl. Diese sagt wie lange soll das Codesegment ausgeführt werden. In den obigen Beispiel lautet das Abbruchbedienung ```anzahlVonZeichen < 10```. Also wird das Befehl ``` Console.WriteLine ``` solange ausgeführt bis das ```anzahlVonZeichen``` kleiner zehn ist. Vielleicht fragst du dich was das Befehlt ```anzahlVonZeichen++``` bedeutet. Es ist eine Abkürkzung von einer Schreibweisse ```anzahlVonZeichen = anzahlVonZeichen + 1```.
+
+Eine Frage: Wieviel _X_ Zeichen werden auf dem Bildschirm in den obigen Beispiel tatsächlich ausgegeben? 
+
+### Aufgabe
+
+Vervollständige das Code hier so dass man ein Alphabet ausgeben kann. Man kann jedoch begrenzen wie viel Zeichen man ausgeben möchte in dem man das 'bis' Zeichen eingibt.
+
+```csharp
+static void Main(string[] args)
+{
+    char[] alphabet = new char[]
+                        {
+                            'a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's', 't', 'u', 'v', 'w', 'x', 'y', 'z', 'ä',  'ö',  'ü', 'ß'
+                        };
+
+    Console.Write("Bis welche Zeichen soll das Alphabet ausgegeben werden (lasse es leer wenn alle Zeichen ausgegeben werden sollten)? ");
+    char zeichen = Console.ReadKey().KeyChar;
+    Console.WriteLine();
+
+    int i = 0;
+
+    //schreibe hier das Codesegment um das Alphabet auszugeben mit hilfe eine while Schleife
+
+    Console.WriteLine();
+}
+
+```
+
+Das Programm könnte so aussehen
+![Alphabet mit eine while Schleife](images/alphabet_with_while.gif)
+
+Es gibt auch eine Fußgesteuerte Variante einer ```while``` Schleife. Die siete so aus:
+
+```csharp
+int anzahlVonZeichen = 0;
+do
+{
+    Console.Write("X");
+    anzahlVonZeichen++;
+}
+while (anzahlVonZeichen < 10)

```

Comment:
**;** missing in the end of the do...while()

Reply:
Oh you are right. I'll repair it.